File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        profit = 0
        L, R = 0, 1

        while R < len(prices):
            if prices[L] < prices[R]:
                profit = max(profit, prices[R] - prices[L])
            else:
                L = R
            R += 1
        
        return profit

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // Little cheat we could do is sort but
        if (nums.size() == 0) {
            return 0;
        }

        sort(nums.begin(), nums.end());
        int length = 1;
        int max = 1;
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] + 1 < nums[i+1]) {
                cout << to_string(nums[i] + 1) << " < " << to_string(nums[i+1]) << endl;
                // sequence bitti, bakalim en buyuk mu
                if (length > max) {
                    max = length;
                }
                length = 1;
            } else if (nums[i] + 1 == nums[i+1]) {
                length++;
            }

            if (length > max) {
                max = length;
            }
        }
        return max;
    }
};
